controllers/mdutil/util.go

```diff
@@ -548,6 +548,15 @@ func NewMSNewReplicas(deployment *clusterv1.MachineDeployment, allMSs []*cluster
 		// Do not exceed the number of desired replicas.
 		scaleUpCount = integer.Int32Min(scaleUpCount, *(deployment.Spec.Replicas)-*(newMS.Spec.Replicas))
 		return *(newMS.Spec.Replicas) + scaleUpCount, nil
+	case clusterv1.OnDeleteMachineDeploymentStrategyType:
+		// Find the total number of machines
+		currentMachineCount := TotalMachineSetsReplicaSum(allMSs)
+		if currentMachineCount >= *(deployment.Spec.Replicas) {
+			// Cannot scale up.
+			return *(newMS.Spec.Replicas), nil
+		}
+		scaleUpCount := *(deployment.Spec.Replicas) - currentMachineCount
+		return *(newMS.Spec.Replicas) + scaleUpCount, nil
```

Comment:
Add more comment around this code explaining the logic a bit more in full form

Reply:
will do!